### ISSUE TITLE
Feature Request - AzureRM Ability to assing Role for Auth

### ISSUE THREAD

**Issue by guitarrapc** [hashicorp/packer#6868]
Hi, I'm wondering if Packer supports MSI Auth then Role Assignment will also needed. (as MSI is only Authentication. Authorization is belong to role assignment.)

https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/tutorial-linux-vm-access-storage#grant-your-vm-access-to-an-azure-storage-container

**Comment by guitarrapc** [comment]
Interesting, how about create custom image from Ubuntu offered image?
My understanding is, create build base image, assign user defined role, then build production image?

So 

```
base Ubuntu image -(packer build without msi)-> base Image -> (packer build with user defined role) -> production image
                                                  ^
                                                  |
                                    assign user defined role
```

**Comment by slikk66** [comment]
Sure, that would work.. There are many examples and workflows. For instance, from a base Ubuntu image, launch it with a user assigned identity that is assigned to roles which allow access to bootstrapping type resources like key vaults, storage containers that have certificates, and let ansible or another provisioner use those credentials during build time to retrieve them. After the build is done, the identity is removed (and not assigned to VMs that will use the image later) so you have safe passage of secrets to a new image.

**Comment by guitarrapc** [comment]
thanks @slikk66 , it make sense.
With above method, it will make us much easier I believe if packer able to assign MSI.

**Comment by boumenot** [comment]
@azr implemented support for MSI in #6161 (PR is #6982).  Does this resolve your issue?

**Comment by guitarrapc** [comment]
@boumenot yes, it resolve my concerns! Great help.
